src/CodewarsResultPrinter.php

```diff
@@ -97,6 +104,7 @@ public function startTestSuite(TestSuite $suite): void
         if (empty($suiteName)) {
             return;
         }
+        $suiteName = $this->prettifier->prettifyTestClass($suiteName);
```

Comment:
Is this correct? `prettifyTestCase($test);` takes a `TestCase`, but `prettifyTestClass` takes a string?

Maybe this is causing the issue you mentioned?

> - Titles of `<DESCRIBE::>` groups of tests generated with `@dataProvider` seem to not respect the annotation (but this might be a bug of my implementation)

Reply:
At least that's how I read it in code of PHPUnit 9.6:
- [`prettifyTestClass(string $className)`](https://github.com/sebastianbergmann/phpunit/blob/9a57283503dea418f17c62f91642d6020425f23f/src/Util/TestDox/NamePrettifier.php#L75C9-L75C9)
- [`prettifyTestCase(TestCase $test)`](https://github.com/sebastianbergmann/phpunit/blob/9a57283503dea418f17c62f91642d6020425f23f/src/Util/TestDox/NamePrettifier.php#L123)

However, the bummer seems to be that PHPUnit 10 seems to change the prettifier: move it to another namespace, and rename the method to [`prettifyTestClassName(string $className)`](https://github.com/sebastianbergmann/phpunit/blob/d77a326390886d870b1b552a2967aa1b7978ed0d/src/Logging/TestDox/NamePrettifier.php#L66). As a result, my reporter will break when PHPUnit gets updated :(

Reply:
> The `::` in `Even Or Odd Test::test Negative Even Numbers` looks weird, though.

I run the suite with original TestDox reporter and it seems that it avoids the weirdness by simply not showing the intermediate groups:

![image](https://github.com/codewars/phpunit-codewars/assets/23709795/b0ddd73f-f3ca-4f0e-9d82-afb41957aaea)

All test cases seem to be flattened, and test cases generated with `@dataProvider` are not grouped together, while CW reporter groups them with a `<DESCRIBE::>`. The title is provided by TestDox prettifier, which apparently does not care much because it's not presented by the TestDox reporter anyway.